payjoin-cli/src/app.rs

```diff
@@ -91,9 +91,11 @@ impl App {
             .danger_accept_invalid_certs(self.config.danger_accept_invalid_certs)
             .build()
             .with_context(|| "Failed to build reqwest http client")?;
+        let req_url = req.0;
+        let req_body = req.1;
         let mut response = client
-            .post(req.url)
-            .body(req.body)
+            .post(req_url)
+            .body(req_body)
```

Comment:
What I'd initially imagined was a change to more easily deconstruct `(req, ctx)` into `((url, body), ctx)`, but I see that one may easily interpret that to instead be deconstructed as in this PR:

```rs
let req_url = req.0;
let req_body = req.1;
```

In hindsight, it actually may make more sense to put context into the request as a named variable so that you have `req.url`, `req.body`, `req.context` and don't have to know about deconstructing the tuple and guess the names. The context is directly related to the particular request anyhow.

I'm going to update the issue to reflect this preference. Thanks for engaging with this so we could come up with a better design.